discord-scripts/incident-report.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  Client,
+  Message,
+  TextChannel,
+  ThreadChannel,
+  VoiceChannel,
+  ActionRowBuilder,
+  ButtonBuilder,
+  ButtonStyle,
+  Events,
+  Interaction,
+} from "discord.js"
+import { Robot } from "hubot"
+// Set to #alarm-trigger channel
+const CHANNEL_ID = "1377183184902688862"
+const { INCIDENT_ROUTING_KEY } = process.env
+
+export default async function incidentReport(
+  discordClient: Client,
+  robot: Robot,
+) {
+  if (!INCIDENT_ROUTING_KEY) {
+    robot.logger.error(
+      "INCIDENT_ROUTING_KEY is not set. Skipping incident report setup.",
+    )
+    return
+  }
+  discordClient.on("messageCreate", async (message: Message) => {
+    if (
+      message.author.bot ||
+      !(
+        message.channel instanceof TextChannel ||
+        message.channel instanceof ThreadChannel ||
+        message.channel instanceof VoiceChannel
+      )
+    ) {
+      return
+    }
+
+    if (message.channel.id !== CHANNEL_ID) return
+
+    if (/code red/i.test(message.content)) {
+      await message.reply({
+        content:
+          "🚨 Code Red detected! (https://youtu.be/WlPTmXi0pVk?feature=shared&t=61)",
+        allowedMentions: { repliedUser: false },
+      })
+    }
+
+    const row = new ActionRowBuilder<ButtonBuilder>().addComponents(
+      new ButtonBuilder()
+        .setCustomId("incident_yes")
+        .setLabel("🚨 Wake Someone Up!")
+        .setStyle(ButtonStyle.Danger),
+      new ButtonBuilder()
+        .setCustomId("incident_no")
+        .setLabel("Don't Trigger")
+        .setStyle(ButtonStyle.Secondary),
+    )
+
+    try {
+      await message.reply({
+        content: `**Before triggering an alert, ask yourself:**  
+- Is this incident truly a Critical or High severity issue?  
+- Could this wait until regular business hours without major impact?  
+- Can I resolve this with existing documentation or procedures?`,
+
+        components: [row],
+        allowedMentions: { repliedUser: false },
+      })
+    } catch (error) {
+      robot.logger.error("❌ Failed to send incident report:", error)
```

Comment:
Non-blocking but in another PR we need to provide them some guidance on what to do next. Sending an email could be a manual way of triggering an alert.